FanGraphSpider.py

```diff
@@ -1,174 +0,0 @@
-                        name = namespot.xpath('a/text()').extract_first()
-                        if plname == name:
-                            url = namespot.xpath('a/@href').extract_first()
-                            break
-                    url = response.urljoin(url)
-                    yield SplashRequest(url, callback=self.home_parse, args=dict(self.SPLASH_ARGS))
-            if 'stolen' in item.xpath('td[6]/text()').extract_first():
-                plname = item.xpath('td[2]/text()').extract_first()
-                # Checks to see if play has already been sent to parser. If not, adds it to list to avoid duplicates.
-                with open('dupcatcher.csv', 'r') as input:
-                    reader = csv.reader(input)
-                    for line in reader:
-                        if [item.xpath('td[6]/text()').extract_first()] == line:
-                            tweeted = True
-                if not tweeted:
-                    with open('dupcatcher.csv', 'a') as output:
-                        writer = csv.writer(output, lineterminator='\n')
-                        line = item.xpath('td[6]/text()').extract_first()
-                        writer.writerow([line])
-                    for namespot in response.xpath('//div[contains(@id, "dg3")]/table/tbody/tr/td[1] | //div[contains(@id, "dg4")]/table/tbody/tr/td[1]'):
-                        name = namespot.xpath('a/text()').extract_first()
-                        if plname == name:
-                            url = namespot.xpath('a/@href').extract_first()
-                            break
-                    url = response.urljoin(url)
-                    yield SplashRequest(url, callback=self.steal_parse, args=dict(self.SPLASH_ARGS))
-
-    # Gathers player stats and constructs tweets for homeruns
-    def home_parse(self, response):
-        p_name = response.xpath('//div[@class="player-info-box-name"]/h1/text()').extract_first()
-        team = response.xpath('//div[@class="player-info-box-name-team"]/a/text()').extract_first()
-        av = response.xpath('//table[@class="rgMasterTable"][1]/tbody/tr[2]/td[17]/text()').extract_first()
-        rstat = response.xpath('//table[@class="rgMasterTable"][1]/tbody/tr[2]/td[7]/text()').extract_first()
-        hr = response.xpath('//table[@class="rgMasterTable"][1]/tbody/tr[2]/td[6]/text()').extract_first()
-        rbi = response.xpath('//table[@class="rgMasterTable"][1]/tbody/tr[2]/td[8]/text()').extract_first()
-        sb = response.xpath('//table[@class="rgMasterTable"][1]/tbody/tr[2]/td[9]/text()').extract_first()
-        obp = response.xpath('//table[@class="rgMasterTable"][1]/tbody/tr[2]/td[18]/text()').extract_first()
-        slg = response.xpath('//table[@class="rgMasterTable"][1]/tbody/tr[2]/td[19]/text()').extract_first()
-        OPS = str(float(obp)+float(slg))

```

Comment:
This block of variables looks the same as the block of variables below in `steal_parse`, so you could extract both of them into a method that looks like this:
```
def parse_response(response):
   return {
     "p_name" : response.xpath('//div[@class="player-info-box-name"]/h1/text()').extract_first(),
     "team" : ...
   }
```
and then do like `stats = parse_response(response)`
. . .